del_events.py

```diff
@@ -44,28 +47,35 @@ def get_credentials():
         flow.user_agent = APPLICATION_NAME
         if flags:
             credentials = tools.run_flow(flow, store, flags)
-        else: # Needed only for compatibility with Python 2.6
+        else:  # Needed only for compatibility with Python 2.6
             credentials = tools.run(flow, store)
         print('Storing credentials to ' + credential_path)
     return credentials
 
+
 def main():
     credentials = get_credentials()
     http = credentials.authorize(httplib2.Http())
     service = discovery.build('calendar', 'v3', http=http)
-    now = datetime.datetime.utcnow().isoformat() + 'Z' # 'Z' indicates UTC time
+    now = datetime.datetime.utcnow().isoformat() + 'Z'  # 'Z' indicates UTC time
     print('Getting the events')
     eventsResult = service.events().list(
-        calendarId='primary', timeMin=now, singleEvents=True,
-        orderBy='startTime', timeMax = "2016-11-20T22:32:00.285773Z").execute()
+        calendarId='primary', timeMin=SEM_BEGIN.strftime('%Y-%m-%dT%H:%M:%S.%fZ'), singleEvents=False, timeMax=END_TERM_BEGIN.strftime('%Y-%m-%dT%H:%M:%S.%fZ'), maxResults=2500).execute()
     events = eventsResult.get('items', [])
+    # print(events)
     if not events:
         print('No upcoming events found.')
     for event in events:
-        if ("Class of " in event["summary"] or "Lab of " in event["summary"]):
-            service.events().delete(calendarId='primary', eventId=event["id"]).execute()
-            print ("Deleted: ", event["summary"], event["start"])
-    print ("Deletion done!")
+        # print(event.get('recurrence'))
+        if 'recurrence' in event and event['recurrence'] in [['RRULE:FREQ=WEEKLY;UNTIL={}'.format(END_TERM_BEGIN.strftime('%Y%m%dT000000Z'))],
+                                                            ['RRULE:FREQ=WEEKLY;UNTIL={}'.format(MID_TERM_BEGIN.strftime('%Y%m%dT000000Z'))],
+                                                            ['RRULE:FREQ=WEEKLY;UNTIL={}'.format(END_TERM_BEGIN.strftime('%Y%m%dT070000Z'))],
+                                                            ['RRULE:FREQ=WEEKLY;UNTIL={}'.format(MID_TERM_BEGIN.strftime('%Y%m%dT080000Z'))]]:
```

Comment:
This `if` clause is pretty long. Might be better to refactor into it's own method: `if is_gyft_event(event): service().delete(event)`

Reply:
Well, if we make it compact as mentioned in your previous review commit, I think it'll be manageable without a function.